Mathlib/Topology/Sheaves/LocalPredicate.lean

```diff
@@ -95,72 +119,292 @@ a `P : LocalPredicate T` consists of:
+def sec : Π b : U, F b := (h b x).choose
+
+lemma pred_sec : P (h.sec x) := (h b x).choose_spec.1.1
+
+lemma sec_apply : h.sec x b = x := (h b x).choose_spec.1.2
+
+lemma sec_eq_of_pred {s : Π b : U, F b} (hs : P s) (b : U) : h.sec (s b) = s :=
+  (isConstantOn_iff.mp h).1 _ _ b (h.pred_sec _) hs (h.sec_apply _)
+
+/-- The monodromy induced by a set of sections constant on `U`. -/
+def monodromy (b b' : U) : F b ≃ F b' where
+  toFun x := h.sec x b'
+  invFun x := h.sec x b
+  left_inv _ := by simp_rw [h.sec_eq_of_pred (h.pred_sec _), h.sec_apply]
+  right_inv _ := by simp_rw [h.sec_eq_of_pred (h.pred_sec _), h.sec_apply]
+
+include h in
+/-- The trivialization induced by a set of sections constant on `U`. -/
+def trivializationOn (b : U) : TrivializationOn P U (F b) where
+  sec := h.sec
+  pred := h.pred_sec
+  bijective _ := (h.monodromy ..).bijective
+
+end IsConstantOn
+
+namespace TrivializationOn
+
+variable {U : Opens B} {ι} (t : TrivializationOn P U ι)
+
+/-- A trivialization of a set of sections over a set `U` induces a trivialization of the `Sigma`
+type over `U`. -/
+def equiv : {f : Σ b, F b // f.1 ∈ U} ≃ U × ι :=
+  .trans (.symm (.trans (.sigmaCongrRight (.ofBijective _ <| t.bijective ·)) .sigmaSubtypeComm))
+    (.sigmaEquivProd ..)
+
+lemma symm_apply_coe (x : U × ι) : t.equiv.symm x = Sigma.mk _ (t.sec x.2 x.1) := rfl
+
+lemma preimage_snd_comp_equiv (i : ι) :
+    Prod.snd ∘ t.equiv ⁻¹' {i} = Subtype.val ⁻¹' Set.range (Sigma.mk _ <| t.sec i ·) :=
+  Set.ext fun f ↦ (Equiv.symm_apply_eq _).trans <| by
+    simp only [Equiv.sigmaSubtypeComm, Equiv.coe_fn_symm_mk,
+      Equiv.ofBijective_apply, Set.mem_preimage, Set.mem_range]
+    refine ⟨fun h ↦ ⟨⟨_, f.2⟩, congr(Sigma.mk _ $h).symm⟩, fun ⟨b, h⟩ ↦ ?_⟩
+    obtain ⟨⟨b', x⟩, hbx⟩ := f
+    cases congr($h.1)
+    exact congr($h.2).symm
+
+end TrivializationOn
+
+end Properties
 
 /-- Continuity is a "local" predicate on functions to a fixed topological space `T`.
 -/
-def continuousLocal (T) [TopologicalSpace T] : LocalPredicate fun _ : X ↦ T :=
+def continuousLocal (T : Type*) [TopologicalSpace T] : LocalPredicate fun _ : X ↦ T :=
   { continuousPrelocal X T with
     locality := fun {U} f w ↦ by
       apply continuous_iff_continuousAt.2
       intro x
-      specialize w x
-      rcases w with ⟨V, m, i, w⟩
+      rcases w x with ⟨V, m, i, w⟩
       dsimp at w
       rw [continuous_iff_continuousAt] at w
       specialize w ⟨x, m⟩
       simpa using (Opens.isOpenEmbedding_of_le i.le).continuousAt_iff.1 w }
 
 /-- Satisfying the inhabited linter. -/
-instance inhabitedLocalPredicate (T) [TopologicalSpace T] :
+instance inhabitedLocalPredicate (T : Type*) [TopologicalSpace T] :
     Inhabited (LocalPredicate fun _ : X ↦ T) :=
   ⟨continuousLocal X T⟩
 
 variable {X T}
 
 /-- The conjunction of two prelocal predicates is prelocal. -/
 def PrelocalPredicate.and (P Q : PrelocalPredicate T) : PrelocalPredicate T where
```

Comment:
Should this be called `inf` as opposed to `and`? Similarly for `LocalPredicate`.

Reply:
If I change the definition to `P.pred ⊓ Q.pred`, probably I should then change the name. You want me to do that and deprecate the old name?